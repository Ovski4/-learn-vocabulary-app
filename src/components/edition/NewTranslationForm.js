import React from 'react';
import { connect } from 'react-redux';
import { translationAdded } from '../../actions/translations';
import { StyleSheet, TextInput, Button, Text, View } from 'react-native';
import Header from '../../components/ui/Header';

const mapDispatchToProps = (dispatch) => ({
    handleTranslationAdded: (translation) => dispatch(translationAdded(translation)),
});

const styles = StyleSheet.create({
    row: {
        flexDirection: 'row'
    },
    word1: {
        flexDirection: 'row',
        alignItems: 'center',
        marginBottom: 5
    },
    word2: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    input: {
        flex: 1,
        backgroundColor: '#eeeeee',
        borderRadius: 2,
        paddingLeft: 5,
        marginLeft: 5
    },
    button: {
        flex: 1,
        justifyContent: 'center',
        margin: 5
    }
});

class NewTranslationForm extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            word1: props.translation.word1,
            word2: props.translation.word2,
            createdAt: props.translation.createdAt
        };
    }

    cannotBeCreated = () => {
        if (this.state.word1.trim().length === 0 ||
            this.state.word2.trim().length === 0
        ) {
            return true;
        }

        return false;
    }

    handleSubmit = () => {
        this.props.handleTranslationAdded({
            word1: this.state.word1,
            word2: this.state.word2,
            createdAt: Date.now()
        });
        this.setState({
            word1: '',
            word2: '',
            createdAt: null
        });
    }

    handleChange = (word, text) => {
        var newWord = {};
        newWord[word] = text;
        this.setState((previousState) => {
            return Object.assign({}, previousState, newWord);
        });
    }

    render() {

        return (
            <View>
                <Header>Add a new translation</Header>
                <View style={styles.row}>
                    <View style={{flex: 4}}>
                        <View style={styles.word1}>
                            <View>
                                <Text>{this.props.labelWord1}</Text>
                            </View>
                            <View style={styles.input}>
                                <TextInput
                                    underlineColorAndroid="transparent"
                                    value={this.state.word1}
                                    onChangeText={(text) => this.handleChange('word1', text)}
                                />
                            </View>
                        </View>
                        <View style={styles.word2}>
                            <View>
                                <Text>{this.props.labelWord2}</Text>
                            </View>
                            <View style={styles.input}>
                                <TextInput
                                    underlineColorAndroid="transparent"
                                    value={this.state.word2}
                                    onChangeText={(text) => this.handleChange('word2', text)}
                                />
                            </View>
                        </View>
                    </View>
                    <View style={styles.button}>
                        <Button disabled={this.cannotBeCreated()} title="Submit" onPress={this.handleSubmit}/>
                    </View>
                </View>
            </View>
        );
    }
}

NewTranslationForm.defaultProps = {
    labelWord1: 'Word 1:',
    labelWord2: 'Word 2:',
    className: 'new-translation',
    translation: {
        word1: '',
        word2: '',
        createdAt: null
    }
};

export default connect(
    null,
    mapDispatchToProps
)(NewTranslationForm);
