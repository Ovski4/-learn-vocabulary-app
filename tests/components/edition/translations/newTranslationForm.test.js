import reducer from '../../../../src/reducers/index';
import { createStore } from 'redux';
import React from 'react';
import NewTranslationForm from '../../../../src/components/edition/translations/newTranslationForm/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('NewTranslationForm component', () => {

    const getStore = (locale) => {
        return createStore(reducer, {
            config: {
                locale: locale
            }
        });
    };

    it('Should render the component with the english locale', () => {
        const rendered = renderer.create(
            <Provider store={getStore('en')}>
                <NewTranslationForm />
            </Provider>
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });

    it('Should render the component with the french locale', () => {
        const rendered = renderer.create(
            <Provider store={getStore('fr')}>
                <NewTranslationForm />
            </Provider>
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });

});
