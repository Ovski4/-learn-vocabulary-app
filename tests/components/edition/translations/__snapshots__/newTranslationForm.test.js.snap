// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTranslationForm component Should render the component with the english locale 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Add a translation
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 5,
        "marginBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 7,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Word 1
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderRadius": 2,
              "flex": 1,
              "marginLeft": 5,
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            defaultValue=""
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "fontSize": 16,
                "height": 32,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Word 2
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderRadius": 2,
              "flex": 1,
              "marginLeft": 5,
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            defaultValue=""
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "fontSize": 16,
                "height": 32,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Tags
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "#eeeeee",
                  "borderRadius": 2,
                  "marginLeft": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 32,
                  }
                }
              >
                <RCTPicker
                  items={
                    Array [
                      Object {
                        "label": "Pick a tag",
                        "textColor": undefined,
                        "value": null,
                      },
                    ]
                  }
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  selectedIndex={0}
                  style={
                    Array [
                      Object {
                        "height": 216,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 2,
          "justifyContent": "center",
          "marginLeft": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={
          Array [
            "disabled",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`NewTranslationForm component Should render the component with the french locale 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Ajouter une traduction
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 5,
        "marginBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 7,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Mot 1
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderRadius": 2,
              "flex": 1,
              "marginLeft": 5,
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            defaultValue=""
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "fontSize": 16,
                "height": 32,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Mot 2
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderRadius": 2,
              "flex": 1,
              "marginLeft": 5,
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            defaultValue=""
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "fontSize": 16,
                "height": 32,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Étiquettes
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "#eeeeee",
                  "borderRadius": 2,
                  "marginLeft": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 32,
                  }
                }
              >
                <RCTPicker
                  items={
                    Array [
                      Object {
                        "label": "Choisissez une étiquette",
                        "textColor": undefined,
                        "value": null,
                      },
                    ]
                  }
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  selectedIndex={0}
                  style={
                    Array [
                      Object {
                        "height": 216,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 2,
          "justifyContent": "center",
          "marginLeft": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={
          Array [
            "disabled",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Valider
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
